source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -265,6 +296,12 @@ Common::Redis::Client::PoolRequest* InstanceImpl::ThreadLocalPool::makeRequestTo
     return nullptr;
   }
 
+  // ensure filter is not removed
```

Comment:
I'm not convinced you need all of these locks everywhere. How can we be making a request with a removed filter? I'm pretty sure you only need it in the construction path for the original set call?

Reply:
The problem is that when the connection pool instance is destroyed the TLS slot is not immediately destroyed.  This leads to the code that cancel the pending requests and closing connections to be delayed as well.  During this delay, a redirection from a pending request would cause a new request to be sent to the new host.

Reply:
OK. Please add more comments around this and also fix format.